.github/workflows/edr-ci.yml

```diff
@@ -157,6 +159,44 @@ jobs:
             **/edr-cache
           key: edr-ts-rpc-cache-v1-${{ matrix.os }}-${{ hashFiles('**/edr-cache/**/*') }}
 
+  test-edr-ts-integration:
+    name: Run TS integration tests (${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
+    needs: check-edr
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
I modified these lines to ensure:
- The matrix is set, as required in the last step
- I added `runs-on` to ensure these tests are run on all operating systems
- I added `fail-fast: fast` to ensure all OSes complete
- I modified `name` to include `matrix.os`
- I added `needs: check-edr` to prevent this from running if that job fails